js/spreadsheet.js

```diff
@@ -36,32 +33,31 @@ var spreadsheet = function(options) {
    * @param {Array} columns
    * @param {Boolean} isFirstRender - defines if it was first render to prepare the data for correct rendering without data changes after changes are made
    */
+
   function prepareData(rows, columns, isFirstRender) {
     var preparedData = rows.map(function(row) {
-      var dataRow = columns.map(function(header, index) {
+      var dataRow = columns.map(function(header) {
         var value = row.data[header];
 
-        if (Array.isArray(value)) {
-          if (arrayColumns.indexOf(header) === -1) {
-            arrayColumns.push(header);
-          }
-
-          // Add double quotes to the string if it contains a comma
-          value = value.map(function(val) {
-            // Stringify value only for the first render for nested arrays
-            if (isFirstRender && value && typeof val !== 'string') {
-              return JSON.stringify(val);
-            }
-
-            return typeof val === 'string' && val.indexOf(',') !== -1 ? '"' + val + '"' : val;
-          }).join(', ');
         // Stringify value only for the first render for nested objects
-        } else if (isFirstRender && value && typeof value === 'object') {
+        if (isFirstRender && typeof value === 'object') {
           if (objColumns.indexOf(header) === -1) {
             objColumns.push(header);
           }
 
-          value = JSON.stringify(value);
+          if (Array.isArray(value)) {
+            // Add double quotes to the string if it contains a comma
```

Comment:
Is this comment still relevant? It doesn't look like we're adding double quotes any more.

Please review all the inline comments in this function to make sure they are relevant and correct.